src/hooks/useAssets.ts

```diff
@@ -0,0 +1,122 @@
+import type { Option, StorageKey } from '@polkadot/types';
+import type { AssetId } from '@polkadot/types/interfaces';
+import type {
+  PalletAssetsAssetAccount,
+  PalletAssetsAssetMetadata,
+  PalletBalancesAccountData,
+} from '@polkadot/types/lookup';
+import { BN } from '@polkadot/util';
+import { useCallback, useState } from 'react';
+
+import { useAccounts } from '@contexts/AccountsContext';
+
+import type { NativeTokenMetadata, TokenBalance, TokenMetadata } from '@helpers/interfaces';
+
+export const useAssets = () => {
+  const { api, activeAccount } = useAccounts();
+  const [tokensMetadata, setTokensMetadata] = useState<TokenMetadata[] | null>(null);
+  const [tokensBalances, setTokensBalances] = useState<TokenBalance[] | null>(null);
+  const [nativeBalance, setNativeBalance] = useState<BN | null>(null);
+  const [nativeMetadata, setNativeMetadata] = useState<NativeTokenMetadata | null>(null);
+
+  const getTokenIds = useCallback(async () => {
+    if (api) {
+      const results: StorageKey<[AssetId]>[] = await api.query.assets.asset.keys();
+      return results.map(({ args: [id] }) => id);
```

Comment:
when I was getting ids for collections and nfts they were provided in an unsorted manner when using .keys
does ordering matter here?

Reply:
The order doesn't matter here. The only place where the order matters is the metadata fetching method